app/validators/sections/section_validator.py

```diff
@@ -176,15 +150,7 @@ def validate_variants(self, block):
             )
 
         for variant in all_variants:
-            when_clause = variant.get("when", [])
-
-            if isinstance(when_clause, list):
-                when_validator = WhenRuleValidator(
-                    when_clause, block["id"], self.questionnaire_schema
-                )
-                self.errors += when_validator.validate()
-
-            elif isinstance(when_clause, dict):
+            if when_clause := variant.get("when", None):
```

Comment:
`.get()` doesn't need `None`, since that is the default anyway.

Reply:
Addressed